beater/server.go

```diff
@@ -126,24 +129,25 @@ func newServer(
 	if err != nil {
 		return server{}, err
 	}
-	httpServer, err := newHTTPServer(logger, info, cfg, tracer, reporter, batchProcessor, agentcfgFetcher, ratelimitStore)
+	httpServer, err := newHTTPServer(logger, info, cfg, tracer, reporter, batchProcessor, fetchReporter, ratelimitStore)
 	if err != nil {
 		return server{}, err
 	}
-	grpcServer, err := newGRPCServer(logger, cfg, tracer, batchProcessor, httpServer.TLSConfig, agentcfgFetcher, ratelimitStore)
+	grpcServer, err := newGRPCServer(logger, cfg, tracer, batchProcessor, httpServer.TLSConfig, fetchReporter, ratelimitStore)
 	if err != nil {
 		return server{}, err
 	}
-	jaegerServer, err := jaeger.NewServer(logger, cfg, tracer, batchProcessor, agentcfgFetcher)
+	jaegerServer, err := jaeger.NewServer(logger, cfg, tracer, batchProcessor, fetchReporter)
 	if err != nil {
 		return server{}, err
 	}
 	return server{
-		logger:       logger,
-		cfg:          cfg,
-		httpServer:   httpServer,
-		grpcServer:   grpcServer,
-		jaegerServer: jaegerServer,
+		logger:        logger,
+		cfg:           cfg,
+		httpServer:    httpServer,
+		grpcServer:    grpcServer,
+		jaegerServer:  jaegerServer,
+		fetchReporter: fetchReporter,
```

Comment:
can we give this a more descriptive name? it fetches and reports what? agentcfgFetchReporter?